tests/foreman/api/test_contentview.py

```diff
@@ -112,6 +112,104 @@ def test_subscribe_system_to_cv(self):
+        lifecycle_env.id = lifecycle_env.create()['id']
+        content_view = entities.ContentView(organization=org.id)
+        content_view.id = content_view.create()['id']
+
+        response = content_view.publish()
+        humanized_errors = response['humanized']['errors']
+        _check_bz_1186432(humanized_errors)
+        self.assertEqual(response['result'], 'success', humanized_errors)
+
+        cvv = entities.ContentViewVersion(
+            id=content_view.read_json()['versions'][0]['id']
+        )
+        response = cvv.promote(lifecycle_env.id)
+        humanized_errors = response['humanized']['errors']
+        _check_bz_1186432(humanized_errors)
+        self.assertEqual(response['result'], 'success', humanized_errors)
+
+        response = content_view.copy(gen_string('alpha', gen_integer(3, 30)))
+        cv_cloned = entities.ContentView(id=response['id'])
+
+        response = cv_cloned.publish()
+        humanized_errors = response['humanized']['errors']
+        _check_bz_1186432(humanized_errors)
+        self.assertEqual(response['result'], 'success', humanized_errors)
+
+        cvv = entities.ContentViewVersion(
+            id=cv_cloned.read_json()['versions'][0]['id']
+        )
+        response = cvv.promote(lifecycle_env.id)
+        humanized_errors = response['humanized']['errors']
+        _check_bz_1186432(humanized_errors)
+        self.assertEqual(response['result'], 'success', humanized_errors)
+
+    def test_cv_clone_within_diff_env(self):
+        """
+        @Test: attempt to create, publish and promote new content
+        view based on existing view but promoted to a
+        different environment
+
+        @Feature: Content Views
+
+        @Assert: Cloned content view can be published and promoted
+        to a different environment as the original content view
+
+        """
+        org = entities.Organization()
+        org.id = org.create()['id']
+        lifecycle_env = entities.LifecycleEnvironment(organization=org.id)
+        lifecycle_env.id = lifecycle_env.create()['id']
+        le_clone = entities.LifecycleEnvironment(organization=org.id)
+        le_clone.id = le_clone.create()['id']
+        content_view = entities.ContentView(organization=org.id)
+        content_view.id = content_view.create()['id']
+
+        response = content_view.publish()
+        humanized_errors = response['humanized']['errors']
+        _check_bz_1186432(humanized_errors)
+        self.assertEqual(response['result'], 'success', humanized_errors)
+
+        cvv = entities.ContentViewVersion(
+            id=content_view.read_json()['versions'][0]['id']
+        )
+        response = cvv.promote(lifecycle_env.id)
+        humanized_errors = response['humanized']['errors']
+        _check_bz_1186432(humanized_errors)
+        self.assertEqual(response['result'], 'success', humanized_errors)
+
+        response = content_view.copy(gen_string('alpha', gen_integer(3, 30)))
```

Comment:
Same nitpick about passing an explicit length to `gen_alpha`. It's fine to pass in a string length if there's some specific reason, but otherwise, I'd just run with whatever the defaults are, thus allowing our code to be a bit more maintainable.